### PR TITLE
Fix semi-sync training with 1GPU per FT replica

### PR DESCRIPTION
**Pull request by bentherien**
When using semi-sync training with a single GPU per replica, no mesh is created in ParallelDims, causing an error when all-reducing the loss.

The slicing error occurs on this line: https://github.com/pytorch/torchtitan/blob/cf30b2902718790cbe91900414c3201b6d7680b0/torchtitan/train.py#L502

This PR prevents this from happening by creating a default mesh when no parallelism is used.

### CODE REVIEW

**Comment by fegin** [comment]
This PR is outdated and not compatible with the current design. Close the PR.
